### PR TITLE
Feature/add moon phase

### PR DESCRIPTION
**Pull request by jnation3406**
This integrates the rise_set changes to incorporate moon phase and adds the valid range on the field. Also snuck in a change to have direct submissions update the last scheduled timestamp since this caused an issue at tlv.

The build tests are failing in here since the ocs-rise-set version isn't deployed yet, but locally they are passing when pointed to the rise-set branch.

### CODE REVIEW

File: observation_portal/observations/viewsets.py [hunk 1]
```diff
@@ -18,6 +18,17 @@
 from observation_portal.common.doc_examples import EXAMPLE_RESPONSES
 
 
+def get_sites_from_request(request):
```
Comment (jnation3406) [inline observation_portal/observations/viewsets.py:21]: The changes in this file are all for the last scheduled timestamp updates for direct submissions.

File: observation_portal/observations/viewsets.py [hunk 2]
```diff
@@ -18,6 +18,17 @@
...
+    sites = set()
```
Comment (mgdaily) [inline observation_portal/observations/viewsets.py:22]: guessing it's a set just to enforce uniqueness?

File: observation_portal/requestgroups/models.py [hunk 3]
```diff
@@ -869,8 +869,9 @@ class Constraints(models.Model):
         help_text='Minimum acceptable angular separation between the target and the moon in decimal degrees'
     )
     max_lunar_phase = models.FloatField(
-        verbose_name='Maximum lunar phase', null=True, blank=True,
-        help_text='Maximum acceptable lunar phase'
+        verbose_name='maximum lunar phase', default=1.0,
+        validators=[MinValueValidator(0.0), MaxValueValidator(1.0)],
+        help_text='Maximum acceptable lunar phase fraction from 0.0 to 1.0 (full moon)'
```
Comment (mgdaily) [inline observation_portal/requestgroups/models.py:874]: Nitpick, maybe add `0.0 (new moon) to 1.0 (full moon)`

Comment (mgdaily) [inline observation_portal/requestgroups/models.py:874]: Nitpick, maybe add `0.0 (new moon) to 1.0 (full moon)`

**Review by mgdaily — APPROVED** [review]
Straight-forward. I like that most of the logic was taken care of in rise-set. LGTM

File: observation_portal/observations/viewsets.py [hunk 4]
```diff
@@ -56,6 +67,16 @@ def perform_create(self, serializer):
...
         return observations_queryset(self.request)
 
+    def create(self, request, *args, **kwargs):
+        """ This sets the last scheduled time on a site when any directly submitted request is submitted for that site
+        """
+        cache_key = 'observation_portal_last_schedule_time'
+        created_obs = super().create(request, args, kwargs)
+        sites = get_sites_from_request(request)
+        for site in sites:
+            cache.set(cache_key + f"_{site}", timezone.now(), None)
```
Comment (jashan-lco) [inline observation_portal/observations/viewsets.py:77]: `f"{cache_key}_{site}"`

**Review by jashan-lco — APPROVED** [review]

**Pull request merged**